lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx

```diff
@@ -0,0 +1,427 @@
+      function() {
+        const { getByRole, rerender } = rtlRender(),
+            inputElement = getByRole('combobox');
+
+        expect(inputElement).toHaveAttribute('aria-autocomplete', 'list');
+
+        rerender(<NxCombobox {...minimalProps} autoComplete={true} />);
+
+        expect(inputElement).toHaveAttribute('aria-autocomplete', 'both');
+      });
+
+  it('sets a completion string of the selected suggestion from matches when `autoComplete` prop is set to true',
+      async function() {
+        const onChange = jest.fn(),
+            { getByRole } = rtlRender({ autoComplete: true,
+              matches: [{ id: '1', displayName: 'Foo' }, { id: '2', displayName: 'Boo' }],
+              onChange}),
+            inputElement = getByRole('combobox');
+
+        inputElement.focus();
+        await userEvent.type(inputElement, 'f');
+        expect(onChange).toBeCalledWith('Foo');
+      });
+
+  it('sets aria-expanded on the input to true when the dropdown displayed which has aria-hidden set to false',
+      async function() {
+        const { getByRole } = rtlRender({ matches: [{ id: '1', displayName: 'Foo' }] }),
+            inputElement = getByRole('combobox'),
+            dropdownElement = getByRole('listbox', { hidden: true });
+
+        expect(inputElement).toHaveAttribute('aria-expanded', 'false');
+        expect(dropdownElement).toHaveAttribute('aria-hidden', 'true');
+        await userEvent.type(inputElement, 'f');
+        expect(inputElement).toHaveAttribute('aria-expanded', 'true');
+        expect(dropdownElement).toHaveAttribute('aria-hidden', 'false');
+      });
+
+  it('calls onChange whenver the input\'s onChange event fires', async function() {
+    const onChange = jest.fn(),
+        { getByRole } = rtlRender({ onChange }),
+        inputElement = getByRole('combobox');
+
+    expect(onChange).not.toHaveBeenCalled();
+    await userEvent.type(inputElement, 'a');
+    expect(onChange).toHaveBeenCalledWith('a');
+  });
+
+  it('calls onSearch whenver the input\'s onChange event fires with a value that differs after trimming, ' +
+    'passing the trimmed value', async function() {
+    const onSearch = jest.fn(),
+        { getByRole } = rtlRender({ value: 'foo', onSearch }),
+        inputElement = getByRole('combobox');
+
+    expect(onSearch).not.toHaveBeenCalled();
+    await userEvent.type(inputElement, ' ');
+    expect(onSearch).not.toHaveBeenCalled();
+    await userEvent.type(inputElement, 'f');
+    expect(onSearch).toHaveBeenCalledWith('foof');
+  });
+
+  it('adds the nx-text-input--long class to the div if set', function() {
+    expect(rtlRender().container.querySelector('.nx-combobox')).not.toHaveClass('nx-text-input--long');
+    expect(rtlRender({ className: 'nx-text-input--long' }).container.querySelector('.nx-combobox'))
+        .toHaveClass('nx-text-input--long');
+  });
+
+  it('adds the nx-text-input--short class to the div if set', function() {
```

Comment:
These can be removed, between the visual tests and the arbitrary extra classname test that I mentioned above this should be adequately covered.

Reply:
Updated here 4029d58f1778f46cf53cdedad783b595e4a9da4e.